core/common/src/main/java/alluxio/network/protocol/databuffer/NioDirectBufferPool.java

```diff
@@ -39,6 +42,23 @@ public static synchronized ByteBuffer acquire(int length) {
     return buffer;
   }
 
+  /**
+   * @param length
+   * @param policy the retry policy to use
+   * @return buffer
+   */
+  public static synchronized ByteBuffer acquire(int length, RetryPolicy policy) {
+    Error cause = null;
+    while (policy.attempt()) {
+      try {
+        return acquire(length);
+      } catch (OutOfMemoryError error) {
+        cause = error;
+      }
+    }
+    throw AlluxioRuntimeException.from("Not enough direct memory allocated to buffer", cause);
```

Comment:
The nit style issue I mentioned is that we can just throw ResourceExhaustedRuntimeException here instead of adding new from method. From method is generally for catching error at very high level(rpc layer)